numpy/core/src/multiarray/typeinfo.c

```diff
@@ -5,9 +5,6 @@
  */
 #include "typeinfo.h"
 
-/* In python 2, this is not exported from Python.h */
-#include <structseq.h>
```

Comment:
for PyPy3 `numpy/core/src/multiarray/typeinfo.c:16:8: error: unknown type name ‘PyStructSequence_Field’`

I'll investigate further.

Reply:
I reverted removing the include because it fails in PyPy.
I opened an issue on pypy to see if this should be fixed in the future but even if it does that won't be fixed for months.
https://bitbucket.org/pypy/pypy/issues/3160/structseqh-missing-from-pythonh

Reply:
Probably better "In PyPy <= 7.3.0", and you can drop the comment if you do
```
#if (defined(PYPY_VERSION_NUM) && (PYPY_VERSION_NUM <= 0x07030000))
/* PyPy issue 3160 */
#include <structseq.h>
#endif
```

Reply:
Comment updated.

Thanks for the informative comments in the PyPy issue (https://bitbucket.org/pypy/pypy/issues/3160/structseqh-missing-from-pythonh)